swagger.yaml

```diff
@@ -1,573 +0,0 @@
-                    - fork_version
-                    - fee_recipient
-        400:
-          description: "Specified fee recipient or withdrawal credentials follow incorrect pattern or amount of ETH is invalid e.g. not a multiple of 32."
-  /v1/ethereum/unstake:
-    post:
-      description: "Unstake x amounts of 32ETH on Ethereum"
-      tags:
-        - staking
-      requestBody:
-        required: true
-        content:
-          "application/json":
-            schema:
-              title: "UnstakeRequest"
-              type: "object"
-              properties:
-                amount:
-                  type: integer
-                  format: integer
-                  description: "Amount of ETH to be unstaked, should be in multiples of 32ETH"
-      responses:
-        202:
-          description: "Accepted - request is being processed in the background"
-        200:
-          description: "Received successful response from Beacon chain for exiting specified amount of ETH. It takes at a minimum 5 epochs (32 minutes) for a validator to exit after initiating a voluntary exit. This number can be much higher depending on how many other validators are queued to exit."
-          content:
-            "application/json":
-              schema:
-                title: ListUnstakeResponse
-                type: array
-                items:
-                  type: "object"
-                  properties:
-                    transaction_hash: 
-                      type: "string"
-                      description: "hash of the signed exit transaction"
-        400:
-          description: "Specified amount of ETH is invalid."
-        409:
-          description: "Specified amount of ETH is not yet in a state available to be unstaked. The amount of ETH should be actively staking, not have been slashed and have been active for at least 256 epochs (~27 hours) since it has been activated to be eligible for unstaking."
-  /v1/ethereum/stats:
-    post:
-      description: "Get rewards and aggregated node statistics for validators hosted by Chorus/Opus"
-      tags:
-        - rewards & performance
-      requestBody:
-        required: true
-        content:
-          "application/json":
-            schema:
-              title: StatsRequest
-              type: "object"
-              properties:
-                from:
-                  type: string
-                  format: date
-                  example: "2019-05-17"
-                  description: |
-                    A starting day for which need to retrieve rewards.
-                    If not sent, the date of registration is used.
-                to:
-                  type: string
-                  format: date
-                  example: "2019-05-17"
-                  description: |
-                    An ending day for which need to retrieve rewards.
-                    By default, assumes it is current day.
-                aggregate_days:
```

Comment:
Frontend has `from` and `to` parameters available for aligning the bins. Backend is going to be super dumb about this.

> don’t do any aggregation, and return the data in the most granular form that it is available? 

This can potentially make super large responses which will be a pity for the http stack to handle. For example, let's say we want rewards for 1 year for 10000 validators. Without doing backend aggregation, this will return 3m 650 000 data points (1 per day for every validator). This will amount to several dozen megabytes in size. Browsers are not well designed to handle that kind of data smoothly.

Reply:
Yes good point, there should be a limit on the size of the date range. We could for v0 limit the size of the range in the backend, and then once it is clear where people run into issues, add endpoints for that specific aggregated data?